go/vt/wrangler/materializer.go

```diff
@@ -941,22 +941,50 @@ func (wr *Wrangler) buildMaterializer(ctx context.Context, ms *vtctldatapb.Mater
 			}
 		}
 	}
-
+	isPartial := false
 	sourceShards, err := wr.sourceTs.GetServingShards(ctx, ms.SourceKeyspace)
 	if err != nil {
 		return nil, err
 	}
+	if len(ms.SourceShards) > 0 {
+		isPartial = true
+		var sourceShards2 []*topo.ShardInfo
+		for _, shard := range sourceShards {
+			for _, shard2 := range ms.SourceShards {
+				if shard.ShardName() == shard2 {
+					sourceShards2 = append(sourceShards2, shard)
+					break
+				}
+			}
+		}
+		sourceShards = sourceShards2
```

Comment:
If sourceShards2 is empty here... is that an error or does it just mean that it's not in fact partial?

Reply:
fixed